.github/actions/slack/action.yaml

```diff
@@ -0,0 +1,91 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+#
+name: Slack notifications
+description: Slack notifications for Bedrock GHA events
+
+inputs:
+  env_name:
+    description: Name of env involved - one of dev, test, stage or prod
+    required: true
+  label:
+    description: Very brief summary of what this notification is for - e.g. 'Integration tests'
+    required: true
+  status:
+    description: Status of message type to be sent, supported values 'info', 'success', 'failure'
+    required: true
+  channel_id:
+    description: Slack channel id to send messages
+    required: true
+  slack_bot_token:
+    description: Slack bot token oauth secret
+    required: true
+  ref:
+    description: Deployment ref
+    required: false
+  message:
+    description: Message body to send
+    required: true
+    default: ""
+outputs:
+  ts:
+    description: Slack timestamp for updating messages
+    value: ${{ steps.slack-notification.outputs.ts }}
+
+runs:
+  using: composite
+
+  steps:
+    - id: set-build-status-color
+      shell: bash
+      run: >-
+        eval "case ${{ inputs.status }} in
+            info)
+                echo 'STATUS_COLOR=#00FFFF';;
+            success)
+                echo 'STATUS_COLOR=#00FF00';;
+            skipped)
+                echo 'STATUS_COLOR=#FF3333';;
+            failure)
+                echo 'STATUS_COLOR=#FF0000';;
+            *)
+                echo 'STATUS_COLOR=##8128a7';;
+        esac" >> $GITHUB_ENV
```

Comment:
Honestly, this little bit of bash was SO fiddly to get working properly in a workflow. I know it can be made more elegant, but unless anyone sees a problem with it, it works so let's leave it as-is 😩